src/main/java/backend/WrappedParam.java

```diff
@@ -0,0 +1,208 @@
+package backend;
+
+/**
+ * We have several types of parameters and we need them to have uniform behavior.
+ * Make sure they have some predictability.
+ *
+ * Derived classes may have more values (all are described below)
+ */
+abstract class WrappedParam {
+    String name;
+    String type;
+    String flag;
+
+    public WrappedParam(String type, String name, String flag){
+        this.type = type;
+        this.name = name;
+        this.flag = flag;
+    }
+
+    /**
+     * Every parameter has a flag.
+     * @return
+     */
+    public String getFlag() {
+        return flag;
+    }
+
+    public void setFlag(String flag) {
+        this.flag = flag;
+    }
+
+    /**
+     * Every parameter has a type.
+     * @return
+     */
+    public String getType(){
+        return type;
+    }
+
+
+    public void setType(String t){
+        type = t;
+    }
+
+    /**
+     * Every parameter, regardless of type, has a name.
+     * @return
+     */
+    public String getName(){
+        return name;
+    }
+
+    public void setName(String n){
+        name = n;
+    }
+}
+
+/**
+ * This represents a SELECTED VALUE of a parameter for our algorithm!
+ * Only use it to feed an algorithm!
+ */
+class WrappedParamFinal extends WrappedParam{
+    String value;
```

Comment:
We're getting a string out of the object... 

Take it up with the Weka devs. :/